capsules/src/virtual_hmac.rs

```diff
@@ -0,0 +1,105 @@
+//! Virtualize the HMAC interface to enable multiple users of an underlying
+//! HMAC hardware peripheral.
+
+use kernel::common::cells::OptionalCell;
+use kernel::common::{List, ListLink, ListNode};
+use kernel::hil::digest;
+use kernel::ReturnCode;
+
+pub struct VirtualMuxHmac<'a, A: kernel::hil::digest::HMACSha256 + digest::Digest<'a>> {
```

Comment:
Why not call this `HmacDevice` to mirror the other virtual_ capsules?

Reply:
Oh I see because of virtual_alarm example. Hmm. I don't have an intuitive idea what a `VirtualMuxX` is, but maybe it makes sense to others.

Reply:
Yeah, I based it on the virtual_alarm. Let me know if you want it changed.